crates/telio-firewall/src/firewall.rs

```diff
@@ -1235,31 +1444,107 @@ impl Firewall for StatefullFirewall {
             .clone()
     }
 
-    fn clear_peer_whitelist(&self) {
-        telio_log_debug!("Clearing firewall peer whitelist");
-        unwrap_lock_or_return!(self.whitelist.write())
-            .peer_whitelist
-            .clear();
+    fn clear_peer_whitelist(&self, permissions: Permissions) {
+        match permissions {
+            Permissions::IncomingConnections => {
+                telio_log_debug!("Clearing firewall incoming connections whitelist");
+                unwrap_lock_or_return!(self.whitelist.write())
+                    .incoming_conn_whitelist
+                    .clear();
+            }
+            Permissions::LocalAreaConnections => {
+                telio_log_debug!("Clearing firewall local area connections whitelist");
+                unwrap_lock_or_return!(self.whitelist.write())
+                    .local_conn_whitelist
+                    .clear();
+            }
+            Permissions::RoutingConnections => {
+                telio_log_debug!("Clearing firewall routing whitelist");
+                unwrap_lock_or_return!(self.whitelist.write())
+                    .routing_whitelist
+                    .clear();
+            }
+        }
+    }
+
+    fn add_to_peer_whitelist(&self, peer: PublicKey, permissions: Permissions) {
+        match permissions {
+            Permissions::IncomingConnections => {
+                telio_log_debug!(
+                    "Adding {:?} peer to firewall incoming connection whitelist",
+                    peer
+                );
+                unwrap_lock_or_return!(self.whitelist.write())
+                    .incoming_conn_whitelist
+                    .insert(peer);
+            }
+            Permissions::LocalAreaConnections => {
+                telio_log_debug!("Adding {:?} peer to firewall local area whitelist", peer);
```

Comment:
With some nice `Display` trait implementation, you could do just:
```
telio_log_debug!("Adding {:?} peer to firewall {} whitelist", peer, permissions);

Reply:
Added display trait in [commit](https://github.com/NordSecurity/libtelio/pull/529/commits/898d738905194518a1d020fb880a3627d19b99d7)